functions/Remove-DbaOrphanUser.ps1

```diff
@@ -149,13 +149,11 @@ function Remove-DbaOrphanUser {
                             Write-Message -Level Verbose -Message "Validating users on database $db."
 
                             if ($User.Count -eq 0) {
-                                #the third validation will remove from list sql users without login. The rule here is Sid with length higher than 16
-                                $User = $db.Users | Where-Object { $_.Login -eq "" -and ($_.ID -gt 4) -and (($_.Sid.Length -gt 16 -and $_.LoginType -eq [Microsoft.SqlServer.Management.Smo.LoginType]::SqlLogin) -eq $false) }
+                                #the third validation will remove from list sql users without login  or mapped to certificate. The rule here is Sid with length higher than 16
```

Comment:
This block of code will always be called, you will never hit the else statement. So all you are doing is increasing the need to maintain the code every time a new login type is created in SQL Server or Azure SQL.

There is no reason to validate a login type because it never has a login, so this will always filter it out: `($User -contains $_.Name)`

Reply:
Correction:

There is no reason to validate if a user has a login and what login type it is because if your user count is greater than `0` you hit your else statement and this validation in your filter is what excludes the user properly: `($User -contains $_.Name)`.

A user in a database that is authenticated with a certificate will never have a login so it will never match the name properly.